subsystems/viamagent/viamagent.go

```diff
@@ -96,72 +101,112 @@ func GetRevision() string {
 }
 
 func Install(logger *zap.SugaredLogger) error {
+	// Check for systemd
+	cmd := exec.Command("systemctl", "--version")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return errw.Wrapf(err, "can only install on systems using systemd, but 'systemctl --version' returned errors %s", output)
+	}
+
 	// Create/check required folder structure exists.
 	if err := agent.InitPaths(); err != nil {
 		return err
 	}
 
-	// Check for systemd
-	_, err := os.Stat(serviceFileDir)
-	if errors.Is(err, fs.ErrNotExist) {
-		return errors.Wrapf(err, "can only install on systems using systemd, but %s is missing", serviceFileDir)
-	}
-	if err != nil {
-		return errors.Wrapf(err, "error getting info for %s", serviceFileDir)
-	}
-
 	// If this is a brand new install, we want to symlink ourselves into place temporarily.
 	expectedPath := filepath.Join(agent.ViamDirs["bin"], subsysName)
 	curPath, err := os.Executable()
 	if err != nil {
-		return errors.Wrap(err, "cannot get path to self")
+		return errw.Wrap(err, "getting path to self")
 	}
 
 	isSelf, err := agent.CheckIfSame(curPath, expectedPath)
 	if err != nil {
-		return errors.Wrap(err, "error checking if installed viam-agent is myself")
+		return errw.Wrap(err, "checking if installed viam-agent is myself")
 	}
 
 	if !isSelf {
 		logger.Infof("adding a symlink to %s at %s", curPath, expectedPath)
-		if err := os.Remove(expectedPath); err != nil && !errors.Is(err, fs.ErrNotExist) {
-			return errors.Wrapf(err, "cannot remove symlink/file at %s", expectedPath)
+		if err := os.Remove(expectedPath); err != nil && !errw.Is(err, fs.ErrNotExist) {
+			return errw.Wrapf(err, "removing symlink/file at %s", expectedPath)
 		}
 		if err := os.Symlink(curPath, expectedPath); err != nil {
-			return errors.Wrapf(err, "cannot install symlink at %s", expectedPath)
+			return errw.Wrapf(err, "installing symlink at %s", expectedPath)
 		}
 	}
 
+	if inSystemdPath(serviceFileDir, logger) {
+		// one-time, remove old /etc based service file when moving to new /usr/local location
+		removeOldServiceFile(logger)
```

Comment:
when we do this we have already written the new service file to disk?

Reply:
Actually, no... just realized that breaks thing if anyone ever tries to use overwrites (the main reason for moving this.) The function itself only runs when the new file isn't yet in place (so it never runs again in the future.) I'll figure something else out.